subt/crash_report.py

```diff
@@ -0,0 +1,38 @@
+"""
+  Keep given buffer of RGBD images until crash happens (detected in accelerometers)
+"""
+import collections
+
+from osgar.node import Node
+from subt.trace import distance3D
+
+
+class CrashReport(Node):
+    def __init__(self, config, bus):
+        super().__init__(config, bus)
+        bus.register("crash_rgbd")
+        size = config.get('size', 100)  # keep N last records
+        self.threshold = config.get('threshold', 100.0)  # i.e. 10g
+        self.buf = collections.deque(maxlen=size)
+
+    def on_rgbd(self, data):
+        self.buf.append(data)
+
+    def on_acc(self, data):
+        vec = [x/1000.0 for x in data]
+        value = abs(distance3D(vec, [0, 0, 0]) - 9.81)
+        if value >= self.threshold:
+            for rec in self.buf:
+                self.publish('crash_rgbd', rec)
```

Comment:
This assumes there will be a single crash report in the whole log. Otherwise we will not be able to tell where one ends and the other starts. This is not necessarily the case, depending on the type of the robot and the acceleration threshold we set.

Options:
1. Output the whole buffer in a single message.
2. Give the crash reports some id.

Reply:
do we really need to separate crash into multiple IDs? When the drones collided there are several peaks and basically this means "collect all data from this period". The issue will be "unfolding" into original timeline, which is potentially swapped for several times.
To dump everything is not good idea as we may just reach the limit